Mathlib/Combinatorics/Additive/NoDoubling.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Patrick Luo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Patrick Luo
+-/
+import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+# Sets with no doubling
+
+This file characterises sets with no doubling (finsets `A` such that `#(A ^ 2) = #A`) as the sets
+which are either empty or translates of a subgroup.
+-/
+
+open MulOpposite MulAction
+open scoped Pointwise RightActions
+
+namespace Finset
+variable {G : Type*} [Group G] [DecidableEq G] {A : Finset G}
+
+/-- A set with no doubling is either empty or the translate of a subgroup. -/
```

Comment:
Let's strengthen the docstring to match the statement more closely: "there exists a subgroup H of G such that every translate of H by an element of A is A itself). Even better would be to give the actual statement which this docstring is saying (as a corollary of this statement): if A is nonempty then there exists `a` with `aH = A`.